drivers/i2c/i2c_ll_stm32.c

```diff
@@ -214,4 +212,5 @@ static int i2c_stm32_init(struct device *dev)
 	return 0;
 }
 
-{{ i2c_stm32.device(data, ['st,stm32-i2c-v1', 'st,stm32-i2c-v2']) }}
+#include "i2c_ll_stm32_device.h"
```

Comment:
I think i2c_ll_stm32_device**s**.h would make more sense.

Then, should we include .c or .h file?
I'm not sure to have any preference, but question should be asked.
@carlescufi ? @galak ?
Any preference?

Reply:
I go back and forth about this, but maybe .c, and adding a comment about so something like:

```
/* Include generated code */
#include "i2c_ll_stm32_device.c"
```

Reply:
I kind of don't like either and would prefer to render the content directly into the driver file instead. 

Like I did here https://github.com/evenl/zephyr/blob/843cfad25ea6c1c6347b77365f8e4d5e171c868d/drivers/spi/spi_ll_stm32.c.jinja2

But I know that I am probably in minority here ;)

Reply:
@evenl 
After thinking about this and discussing it with people, I really dislike the templated C file. I would like my `.c` file to be a `.c` file, clean of templating and containing only `.c` code.

Reply:
Ok, I am anyway generating .h file as you requested in the first place.